src/storage/storage.cpp

```diff
@@ -433,394 +492,515 @@ Storage::ReturnCode Storage::Run(int max_wait)
-    geometry_input_stream.read((char *)&temporary_value, sizeof(unsigned));
-    BOOST_ASSERT(temporary_value ==
-                 shared_layout_ptr->num_entries[SharedDataLayout::GEOMETRIES_INDEX]);
+        // load the edges of the search graph
+        QueryGraph::EdgeArrayEntry *graph_edge_list_ptr =
+            layout_ptr->GetBlockPtr<QueryGraph::EdgeArrayEntry, true>(memory_ptr,
+                                                                      DataLayout::GRAPH_EDGE_LIST);
 
-    if (shared_layout_ptr->GetBlockSize(SharedDataLayout::GEOMETRIES_INDEX) > 0)
-    {
-        geometry_input_stream.read(
-            (char *)geometries_index_ptr,
-            shared_layout_ptr->GetBlockSize(SharedDataLayout::GEOMETRIES_INDEX));
+        io::readHSGR(hsgr_input_stream,
+                     graph_node_list_ptr,
+                     hsgr_header.number_of_nodes,
+                     graph_edge_list_ptr,
+                     hsgr_header.number_of_edges);
+        hsgr_input_stream.close();
     }
-    NodeID *geometries_node_id_list_ptr = shared_layout_ptr->GetBlockPtr<NodeID, true>(
-        shared_memory_ptr, SharedDataLayout::GEOMETRIES_NODE_LIST);
-
-    geometry_input_stream.read((char *)&temporary_value, sizeof(unsigned));
-    BOOST_ASSERT(temporary_value ==
-                 shared_layout_ptr->num_entries[SharedDataLayout::GEOMETRIES_NODE_LIST]);
 
-    if (shared_layout_ptr->GetBlockSize(SharedDataLayout::GEOMETRIES_NODE_LIST) > 0)
+    // ram index file name
     {
-        geometry_input_stream.read(
-            (char *)geometries_node_id_list_ptr,
-            shared_layout_ptr->GetBlockSize(SharedDataLayout::GEOMETRIES_NODE_LIST));
-    }
-    EdgeWeight *geometries_fwd_weight_list_ptr = shared_layout_ptr->GetBlockPtr<EdgeWeight, true>(
-        shared_memory_ptr, SharedDataLayout::GEOMETRIES_FWD_WEIGHT_LIST);
+        char *file_index_path_ptr =
+            layout_ptr->GetBlockPtr<char, true>(memory_ptr, DataLayout::FILE_INDEX_PATH);
+        // make sure we have 0 ending
+        std::fill(file_index_path_ptr,
+                  file_index_path_ptr + layout_ptr->GetBlockSize(DataLayout::FILE_INDEX_PATH),
+                  0);
+        auto absolute_file_index_path = boost::filesystem::absolute(config.file_index_path);
+        std::copy(absolute_file_index_path.string().begin(),
+                  absolute_file_index_path.string().end(),
+                  file_index_path_ptr);
+    }
+
+    // Name data
+    {
+        /* To be replaced by io loading code */
+        boost::filesystem::ifstream name_stream(config.names_data_path, std::ios::binary);
+        unsigned name_blocks = 0;
+        name_stream.read((char *)&name_blocks, sizeof(unsigned));
+        unsigned number_of_chars = 0;
+        name_stream.read((char *)&number_of_chars, sizeof(unsigned));
+        /* END to be replaced */
+
+        // Loading street names
+        unsigned *name_offsets_ptr =
```

Comment:
not only the auto ptrs but also this ptr should be possible to convert to `const unsigned *`